MDANSE_GUI/Src/MDANSE_GUI/Widgets/PlotSettingsDialog.py

```diff
@@ -0,0 +1,339 @@
+
+    Returns
+    -------
+    Any
+        Typically a string, list[float], list[str] or None
+    """
+    param_str = str(param_str).replace("\\", "")
+    param_str = param_str.replace("'", "")
+    if "[" in param_str:
+        toks = param_str.strip("[]").split(",")
+        try:
+            result = [float(x) for x in toks]
+        except ValueError:
+            result = toks
+    elif param_str == "None":
+        return None
+    else:
+        result = param_str
+    return result
+
+
+def convert_to_string(param_str: str) -> str | None:
+    """Convert a Python string to a matplotlib-friendly representation.
+
+    In matplotlib settings, lists should not have brackets around them,
+    and strings have no quotes.
+
+    Parameters
+    ----------
+    param_str : str
+        String obtained from a Python object (i.e. a value from rcParams)
+
+    Returns
+    -------
+    str | None
+        a simplified string for matplotlib, or None is input is "None"
+    """
+    result = str(param_str).replace("\\", "")
+    result = result.replace("'", "")
+    result = result.strip("[]")
+    if "['']" in result:
+        result = result.replace["['']", "[]"]
+    return result
+
+
+EXPOSE_KEYS = [
+    "legend.fontsize",
+    "legend.borderpad",
+    "legend.labelspacing",
+    "xtick.labelsize",
+    "ytick.labelsize",
+    "axes.titlesize",
+    "axes.labelsize",
+    "axes.grid",
+    "font.size",
+]
+
+BAD_KEYS = [
+    "axes.prop_cycle",
+    "lines.dash_capstyle",
+    "lines.dash_joinstyle",
+    "lines.marker",
+    "lines.solid_capstyle",
+    "lines.solid_joinstyle",
+]
+
+ALL_MAJOR_KEYS = {key.split(".")[0] for key in rcParams}
+PREFERRED_SORT_KEYS = [
+    "axes",
+    "grid",
+    "figure",
+    "font",
+    "lines",
+    "legend",
+    "savefig",
+    "xtick",
+    "ytick",
+]
+SORT_MAJOR_KEYS = [key for key in sorted(PREFERRED_SORT_KEYS) if key in ALL_MAJOR_KEYS]
+
+
+class PlotSettingsModel(QStandardItemModel):
+    """Interface between matplotlib's rcParams and QTreeView."""
+
+    plots_need_updating = Signal()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._row_lookup = {}
+        self.populate_model(rcParams)
+        self.itemChanged.connect(self.update_single_value)
+        self.setHorizontalHeaderLabels(["Parameters", "Values", "Default"])
+
+    def populate_model(self, par_dict: dict[str, str]):
+        """Put entries from rcParams into the Qt model.
+
+        Skips a few entries which are difficult to parse, because
+        they require instances of specific matplotlib classes.
+
+        Parameters
+        ----------
+        par_dict : dict[str, str]
+            key:value pairs from rcParams, converted to string.
+        """
+        for key, value in par_dict.items():
+            if key in BAD_KEYS:
+                continue
+            left_item = QStandardItem(str(key))
+            right_item = QStandardItem(str(value))
```

Comment:
Given the dictionary is typed to be `str, str` this is assumed to be safe and a user error.
```suggestion
            left_item = QStandardItem(key)
            right_item = QStandardItem(value)
```